ignite/handlers/checkpoint.py

```diff
@@ -708,7 +709,7 @@ def _save_func(self, checkpoint: Mapping, path: str, func: Callable, rank: int =
             else:
                 if tmp is not None:
                     tmp.close()
-                    os.rename(tmp.name, path)
+                    shutil.move(tmp.name, path)
```

Comment:
Why this in the PR?

Reply:
Windows has `FileExistError` with `os.rename`, so i changed to `shutil.move`.

Reply:
It means that was not functional on macOS and windows ? Did you state that introducing a new test ?

Reply:
I added [these extra tests](https://github.com/pytorch/ignite/pull/1808/files#diff-bd52578edbfac3a9213886eb448a36cf7c13ad3c95110279ccf1d76c9b4a35a7R200-R217) to cover missing parts and `test_setup_common_training_handlers` test passed on mac and linux, failed on windows resulting an error of `FileExistError` with `os.rename`.
So I changed to `shutil.move`

Reply:
It is functional, what happens here is that in a single test @ydcjeff saves 3 times the same file. Normally, we do not overwrite existing files. With this change we will overwrite without any issue...